Lmr 55 feedback about instrument and owner

Daria-pugacheva opened the pull request:
Доработан функционал по отправке отзывов об инструменте и его владельце (по отдельности  и вместе) в соответствии с логикой на бэке, которая добавлена Артемом в ветке LMR-12
1. В InstrumentInfoDto написала TODO с просьбой добавить поле с айдишником владельца. Очень надо для реализации функционала отправки отзывов. Соответственно, в ревьюеры добавила Алексея, т.к. он занимался, вроде, конвертерами для Instrument. Наверное, ему сподручнее будет эту задачу решать, чтобы везде, как надо сделать коррективы
2. Пока не запускала приложение для проверки функциональности, так что, как и в ветке LMR-55-feedbackaboutRenter, пока под сомнением корректность ввода полей ng-model с необходимыми айдишниками и отправки хэдера на бэк
3. Думаю, что хорошо бы еще прямо на фронте добавить валидацию на поле ввода оценки (grade). Но я это пока не реализовала.

Daria-pugacheva committed:
Накидала функционал отправки отзывов владельца об арендаторе с учетом того, что на бэке по этому функционалу предложил сделать Артем.
1. Сделана отдельная страница для написания отзывов.
Пока нет возможности запустить посмотреть, как функционирует, сомневаюсь в двух моментах:
 - корректно ли я заполняю по умолчанию два поля модели - это строки 4 и 5 в feedback-about-renter-page.js
 - корректно ли отправляю нужных хэдер "about"  - строки 11-13 в feedback-about-renter-page.js
2. Поправила в соответствии с логикой методы в файлах с tool-history
3. Оставила TODO запрос в InstrumentRentDto, чтобы в нем появилось еще поле с айдишником арендатора (он нам нужен для отправки отзывов по предложенной Артемом логике)
4. И в UserController тоже оставила предложение сделать метод, отдающий данные о юзере по айдишнику
5. В создала глобальную переменную $rootScope.myUserIdFromProfile для реализации  отправок отзывов теперь сохраняю айдишник текущего пользователя.

Daria-pugacheva committed:
Merge branch 'LMR-55-feddbackaboutrenter' into LMR-55-feedbackAboutInstrumentAndOwner

# Conflicts:
#	source/core/src/main/resources/static/profile/profile.html
#	source/core/src/main/resources/static/profile/profile.js

Daria-pugacheva committed:
Доработан функционал по отправке отзывов об инструменте и его владельце (по отдельноти  и вместе) в соответствии с логикой на бэке, которая добавлена Артемом в ветке LMR-12
1. В InstrumentInfoDto написала TODO с просьбой добавить поле с айдишником владельца. Очень надо для реализации функционала отправки отзывов.
2. Пока не запускала приложение для проверки функциональности, так что, как и в ветке LMR-55-feedbackaboutRenter, пока под сомнением корректность ввода полей ng-model с необходимыми айдишниками и отправки хэдера на бэк
3. Думаю, что хорошо бы еще прямо на фронте добавить валидацию на поле ввода оценки (grade). Но я это пока не реализовала.

Daria-pugacheva committed:
Небольша доработка : в profile.js добавила $rootScope.myUserIdFromProfile

Daria-pugacheva committed:
Update source/core/src/main/resources/static/feedback-about-renter-page/feedback-about-renter-page.js

Co-authored-by: Nikolay Balashov <46228837+siporqueno@users.noreply.github.com>

Daria-pugacheva committed:
Update source/core/src/main/resources/static/feedback-about-renter-page/feedback-about-renter-page.js

Co-authored-by: Nikolay Balashov <46228837+siporqueno@users.noreply.github.com>

Daria-pugacheva committed:
Merge branch 'develop' into LMR-55-feddbackaboutrenter

Daria-pugacheva committed:
Merge remote-tracking branch 'origin/develop' into LMR-55-feddbackaboutrenter

# Conflicts:
#	source/core/src/main/resources/static/profile/profile.html
#	source/core/src/main/resources/static/profile/profile.js

Daria-pugacheva committed:
Небольша доработка : в profile.js добавила $rootScope в используемые зависимости.
И смержила ветку с текущей develop для удобства

Daria-pugacheva committed:
Merge remote-tracking branch 'origin/LMR-55-feddbackaboutrenter' into LMR-55-feddbackaboutrenter

# Conflicts:
#	source/core/src/main/resources/static/profile/profile.html
#	source/core/src/main/resources/static/profile/profile.js

Daria-pugacheva committed:
Крошечные правки для тестирования работоспособности (уберу их, когда проверю)

Daria-pugacheva committed:
Merge remote-tracking branch 'origin/develop' into LMR-55-feedbackAboutInstrumentAndOwner

Daria-pugacheva committed:
Смержила ветку со свежей develop +  поправила названия header-ов в функциях отправки отзывов и заменила в этих функциях param на data

tatiananosova commented on source/core/src/main/resources/static/profile/profile.html:
```diff
@@ -1,17 +1,12 @@
 <div class="container">
```
Эта страница рабочая, непонятно зачем изменять её на старую версию с закомментированным кодом?

Daria-pugacheva replied:
Видимо, у меня какие-то накладки в этот раз с мержем  девелопа в эти ветки.  Да profile.js в итоге будет с рабочим кодом (закомментирован пока только тот, что грузит заказы). В очередном коммите поправлю

tatiananosova commented on source/core/src/main/resources/static/profile/profile.js:
```diff
@@ -1,5 +1,5 @@
-angular.module('tools').controller('profileController', function ($scope, $http, $location) {
-    const contextPath = 'http://localhost:8890/let-me-rent/';
+angular.module('tools').controller('profileController', function ($scope,rootScope, $http,  $location) {
+     const contextPath = 'http://localhost:8890/let-me-rent/api/v1/instruments/';
```
Запросы в файле используют разный пасс, поэтому тут нужно оставить изначальный вариант, не все они идут через /instruments

Daria-pugacheva replied:
Согласна, что должно быть  http://localhost:8890/let-me-rent/ - тоже постоянно в мерже вылезает эта строка с инструментами и я ее убираю, видимо, в очередной подгрузке не убралась. Исправлю

tatiananosova commented:
Для меня не понятно, в предыдущем пул-реквесте почти такие же изменения, такое ощущение, что они унаследованы один от другого. Наверное, эту ветку нужно мержить не в девелоп, а в LMR-55-feddbackaboutrenter

Daria-pugacheva committed:
В index.js добавлены обратно пропавшие две скобки в троке 54

Daria-pugacheva committed:
Merge branch 'LMR-55-feddbackaboutrenter' into LMR-55-feedbackAboutInstrumentAndOwner

# Conflicts:
#	source/core/src/main/resources/static/profile/profile.html
#	source/core/src/main/resources/static/profile/profile.js

Daria-pugacheva commented:
Да, я , видимо, подзапуталась с этим перекрестным мержем.  Сейчас хочу сделать одну ветку общую на LMR-55 задачу всю

Daria-pugacheva committed:
Залила в эту ветку ветвь с отзывами об арендаторе + сделала пока ручное заполнение двух полей в ng-model произвольными цифрами с id инструмента, аредатора, владельца и автора отзыва.
Есть желание продолжить работать над задачей 55 в этой одной ветке.
Пока у меня вот прямо лютый затык с вопросом, как передавать данные про  id инструмента, аредатора, владельца и автора отзыва в соответствующие поля ng-model. Может, кто подскажет? Я пробовала и так и эдак, не получается. Вот что пыталась предпринять:
1. $scope.tool_comment.instrumentId = $routeParams.toolId; - это я, когда наброски делала, нафантазировала, но такой вариант не рабочий.
2. В самой html добваить в Input-поле атрибут valuе чтобы он по умолчанию заполнялся нашими данными, не получается (я так понимаю, поскольку у нас там уже есть ng-model  и она перекрывает)
3. На странице js просто создала объекты типа $scope.tool_comment={userId: "1", instrumentId: "2"}; - но тут только ручной ввод именно значений и не нашла как в json-нину передать нужные мне переменные $rootScope.myUserIdFromProfile, $routeParams.toolId и т.д. вместо цифр
4. Была еще мысль уж отправлять с фронта запрос на бэк, в котором передавать два айдишника в PathVariable и ответом как бы эхо получать обхект с названиями полей, как нужно нам в модели, заполенными этими значениями и тогда как бы мы с бэка полчаем модель с двумя полями. Но это какая-то дичь... Все-таки хотелось бы прямо на фронте засовывать в ng-model известные нам поля.

Daria-pugacheva committed:
Переименовала ветку, чтобы было понятно, что в ней все варианты отправки комментариев
